src/math.cpp

```diff
@@ -19,4 +27,95 @@ auto abs(const CallContext& ctx) -> Value {
         throw std::runtime_error("bad argument #1 to abs (number expected, got " + x.type() + ")");
     }
 }
+
+auto acos(const CallContext& ctx) -> Value {
+    auto x = ctx.arguments().get(0);
+    auto res = minilua::to_number(ctx);
+
+    if (res != Nil()) {
+        auto num = get<Number>(res);
+
+        if (num.value >= -1 && num.value <= 1) {
+            return Value(std::acos(num.value));
+        } else {
+            return Value("nan");
+        }
+    } else {
+        throw std::runtime_error("bad argument #1 to acos (number expected, got " + x.type() + ")");
+    }
+}
+
+auto asin(const CallContext& ctx) -> Value {
+    auto x = ctx.arguments().get(0);
+    auto res = minilua::to_number(ctx);
+
+    if (res != Nil()) {
+        auto num = get<Number>(res);
+
+        if (num.value >= -1 && num.value <= 1) {
+            return Value(std::asin(num.value));
+        } else {
+            return Value("nan");
+        }
+    } else {
+        throw std::runtime_error("bad argument #1 to asin (number expected, got " + x.type() + ")");
+    }
+}
+
+auto atan(const CallContext& ctx) -> Value {
+    // unimplemented
+}
+
+auto ceil(const CallContext& ctx) -> Value {
+    auto x = ctx.arguments().get(0);
+    auto res = minilua::to_number(ctx);
+
+    if (res != Nil()) {
+        auto num = get<Number>(res);
+
+        return Value(std::ceil(num.value));
+    } else {
+        throw std::runtime_error("bad argument #1 to ceil (number expected, got " + x.type() + ")");
+    }
+}
+
+auto cos(const CallContext& ctx) -> Value {
+    auto x = ctx.arguments().get(0);
+    auto res = minilua::to_number(ctx);
+
+    if (res != Nil()) {
+        auto num = get<Number>(res);
+
+        return Value(std::cos(num.value));
+    } else {
+        throw std::runtime_error("bad argument #1 to cos (number expected, got " + x.type() + ")");
+    }
+}
+
+auto deg(const CallContext& ctx) -> Value {
+    auto x = ctx.arguments().get(0);
+    auto res = minilua::to_number(ctx);
+
+    if (res != Nil()) {
+        auto num = get<Number>(res);
+        double deg = num.value * 180 / boost::math::constants::pi<double>();
```

Comment:
don't use boost so we can remove it as a dependency once we throw out the old parser/interpreter code.

Pi should be available in `math.h`. See: https://stackoverflow.com/a/1727896
Or you can just copy the constant from boost.